src/views/projectExplorer/project.ts

```diff
@@ -14,4 +29,68 @@ export default class Project extends TreeItem {
     this.contextValue = Project.contextValue;
     this.description = description;
   }
+
+  async getChildren(): Promise<ProjectExplorerTreeItem[]> {
+    let items: ProjectExplorerTreeItem[] = [];
+
+    const iProject = ProjectManager.get(this.workspaceFolder);
+
+    const ibmi = getInstance();
+    const deploymentDirs = ibmi?.getStorage().getDeployment()!;
+    const localDir = this.resourceUri?.path!;
+    const remoteDir = deploymentDirs[localDir];
+
+    // First load the IFS browser stuff
+    if (remoteDir) {
+      items.push(new IFSDirectory(this.workspaceFolder, remoteDir, `Source`));
+    } else {
+      items.push(new ErrorItem(this.workspaceFolder, `Source`, {
+        description: `Please configure remote directory.`,
+        command: {
+          command: `code-for-ibmi.setDeployLocation`,
+          title: `Set deploy location`,
+          arguments: [{}, this.resourceUri]
+        }
+      }));
+    }
+
+    // Then load the variable specific stuff
+    await iProject?.read();
+
+    const hasEnv = await iProject?.projectFileExists('.env');
+    if (hasEnv) {
+      let unresolvedVariableCount = 0;
+
+      const possibleVariables = iProject?.getVariables();
+      const actualValues = await iProject?.getEnv();
+      if (possibleVariables && actualValues) {
+        unresolvedVariableCount = possibleVariables.filter(varName => !actualValues[varName]).length;
+      }
+
+      items.push(new Variables(this.workspaceFolder, unresolvedVariableCount));
+
+    } else {
+      items.push(new ErrorItem(this.workspaceFolder, `Variables`, {
+        description: `Please configure environment file.`,
+        command: {
+          command: `vscode-ibmi-projectmode.createEnv`,
+          arguments: [this.workspaceFolder],
+          title: `Create project .env`
+        }
+      }));
+    }
+
+    items.push(new ObjectLibraries(this.workspaceFolder));
+
+    for await (const extensibleChildren of Project.callBack) {
+      this.extensibleChildren.push(...await extensibleChildren(iProject!));
```

Comment:
Something to consider is how we are going to handle errors from the `extensibleChildren` callback. If `extensibleChildren` crashes here, then nothing will be returned and the view will be blank.

Reply:
Added a try catch around the callback